asv_bench/benchmarks/pandas/join_merge.py

```diff
@@ -0,0 +1,261 @@
+    # GH 6329
+    def setup(self):
+        date_index = date_range("01-Jan-2013", "23-Jan-2013", freq="T")
+        daily_dates = date_index.to_period("D").to_timestamp("S", "S")
+        self.fracofday = date_index.values - daily_dates.values
+        self.fracofday = self.fracofday.astype("timedelta64[ns]")
+        self.fracofday = self.fracofday.astype(np.float64) / 86_400_000_000_000
+        self.fracofday = Series(self.fracofday, daily_dates)
+        index = date_range(date_index.min(), date_index.max(), freq="D")
+        self.temp = Series(1.0, index)[self.fracofday.index]
+
+    def time_join_non_unique_equal(self):
+        self.fracofday * self.temp
+
+
+class Merge:
+
+    params = [True, False]
+    param_names = ["sort"]
+
+    def setup(self, sort):
+        N = 10000
+        indices = tm.makeStringIndex(N).values
+        indices2 = tm.makeStringIndex(N).values
+        key = np.tile(indices[:8000], 10)
+        key2 = np.tile(indices2[:8000], 10)
+        self.left = DataFrame(
+            {"key": key, "key2": key2, "value": np.random.randn(80000)}
+        )
+        self.right = DataFrame(
+            {
+                "key": indices[2000:],
+                "key2": indices2[2000:],
+                "value2": np.random.randn(8000),
+            }
+        )
+
+        self.df = DataFrame(
+            {
+                "key1": np.tile(np.arange(500).repeat(10), 2),
+                "key2": np.tile(np.arange(250).repeat(10), 4),
+                "value": np.random.randn(10000),
+            }
+        )
+        self.df2 = DataFrame({"key1": np.arange(500), "value2": np.random.randn(500)})
+        self.df3 = self.df[:5000]
+
+    def time_merge_2intkey(self, sort):
+        merge(self.left, self.right, sort=sort)
+
+    def time_merge_dataframe_integer_2key(self, sort):
+        merge(self.df, self.df3, sort=sort)
+
+    def time_merge_dataframe_integer_key(self, sort):
+        merge(self.df, self.df2, on="key1", sort=sort)
+
+    def time_merge_dataframe_empty_right(self, sort):
+        merge(self.left, self.right.iloc[:0], sort=sort)
+
+    def time_merge_dataframe_empty_left(self, sort):
+        merge(self.left.iloc[:0], self.right, sort=sort)
+
+    def time_merge_dataframes_cross(self, sort):
+        merge(self.left.loc[:2000], self.right.loc[:2000], how="cross", sort=sort)
+
+
+class I8Merge:
+
+    params = ["inner", "outer", "left", "right"]
+    param_names = ["how"]
+
+    def setup(self, how):
+        low, high, n = -1000, 1000, 10**6
+        self.left = DataFrame(
+            np.random.randint(low, high, (n, 7)), columns=list("ABCDEFG")
+        )
+        self.left["left"] = self.left.sum(axis=1)
+        self.right = self.left.sample(frac=1).rename({"left": "right"}, axis=1)
+        self.right = self.right.reset_index(drop=True)
+        self.right["right"] *= -1
+
+    def time_i8merge(self, how):
+        merge(self.left, self.right, how=how)
+
+
+class MergeCategoricals:
```

Comment:
I'd leave 1-2 cases with categorical but again not sure whether they will trigger some unique execution paths in Modin

Reply:
I've definitely seen a couple of places where there is a separate logic for categories, let's leave it for now.

Reply:
Okay